src/main.rs

```diff
@@ -0,0 +1,160 @@
+#[macro_use]
+extern crate chan;
+extern crate chan_signal;
+
+#[macro_use]
+extern crate clap;
+
+extern crate irc;
+
+use std::io::prelude::*;
+use std::io::BufReader;
+
+use irc::client::prelude::*;
+use std::default::Default;
+use std::str::FromStr;
+
+use chan_signal::Signal;
+use std::thread::spawn;
+
+/// Actions that the IRC server can take
+enum Action {
+    Quit,
+    Join,
+}
+
+fn main() {
+    // Catch signals we expect to exit cleanly from
+    let signal = chan_signal::notify(&[Signal::INT, Signal::TERM, Signal::PIPE]);
+
+    let matches = clap_app!
+        (hashpipe =>
+         (version: "0.1")
```

Comment:
Cool trick @ianweller pointed out for version and author: https://github.com/euank/synapse-password-reset/blob/a7da24858cbcaf2af00fd3a8b82e1581e4d157f9/src/main.rs#L101-L102